compiler/src/dotty/tools/dotc/reporting/messages.scala

```diff
@@ -1846,15 +1862,28 @@ class FailureToEliminateExistential(tp: Type, tp1: Type, tp2: Type, boundSyms: L
         |are only approximated in a best-effort way."""
 }
 
-class OnlyFunctionsCanBeFollowedByUnderscore(tp: Type)(using Context)
+class OnlyFunctionsCanBeFollowedByUnderscore(tp: Type, tree: untpd.PostfixOp)(using Context)
   extends SyntaxMsg(OnlyFunctionsCanBeFollowedByUnderscoreID) {
   def msg(using Context) = i"Only function types can be followed by ${hl("_")} but the current expression has type $tp"
   def explain(using Context) =
     i"""The syntax ${hl("x _")} is no longer supported if ${hl("x")} is not a function.
         |To convert to a function value, you need to explicitly write ${hl("() => x")}"""
-}
 
-class MissingEmptyArgumentList(method: String)(using Context)
+  override def actions(using Context) =
+    import scala.language.unsafeNulls
+    val untpd.PostfixOp(qual, Ident(nme.WILDCARD)) = tree: @unchecked
+    List(
+      CodeAction(title = "Rewrite to function value",
+        description =  java.util.Optional.empty(),
+        patches = List(
+          ActionPatch(SourcePosition(tree.source, Span(tree.span.start)), "(() => "),
+          ActionPatch(SourcePosition(tree.source, Span(qual.span.end, tree.span.end)), ")")
```

Comment:
This duplicates the logic with `patch(...)` calls at the use-site of OnlyFunctionsCanBeFollowedByUnderscore, I think having the actions in the message like this makes a lot of sense but we need to drop the old `patch` mechanism at the same time otherwise we'll be in a weird situation where -rewrite and IDEs each do their own thing.

Reply:
Good call. So what I did here was that I ended up making the patches inside of the message like I was before, but now instead of putting `patch`es together like this outside of the message, we instead create the message and then just grab the patch out of the message. This still isn't idea and something we talked about this morning was completely refactoring the way that patches are applied by waiting until the reporting happens and during that time, check if the message that is being reported has an action, and if so, apply it, which contains the patch. However, thinking about this a bit more this afternoon @smarter, one limitation of that approach is that many places where we do patches, messages don't exist. In fact, the vast majority of the places that we call `patch`, there is no `Message` for that. So in order for that to work we'd really want to create a unique `Message` for it that would hold that actions/patch.

Reply:
I guess we could keep `patch` around for formatting changes (braces <-> indent) since flooding the console with messages for those changes wouldn't be useful.